### PR TITLE
feat(utils/string1): add isStringEqual function 

### LINKED ISSUE #29: isStringEqual
### `isStringEqual`

Checks if two provided strings are equal.

`src/utils/string/isStringEqual.ts`

**Parameters**:
- `str1: string`: The first string to compare.
- `str2: string`: The second string to compare.

**Returns**:
- `boolean`: Returns `true` if both strings are equal, otherwise returns `false`.

**Example**:
```ts
isStringEqual('Hello', 'Hello'); // true
isStringEqual('Hello', 'World');  // false
isStringEqual('Hello', 'hello');  // false (case-sensitive)

### PR DESCRIPTION
**Pull request by nikkivaddepelli**
closes #29 

## Summary by CodeRabbit

- **New Features**
	- Introduced the `isStringEqual` function to compare two strings, including handling of `null` and `undefined` values, with options for ignoring whitespace and case.
- **Tests**
	- Added a test suite for the `isStringEqual` function, including cases for identical strings, different strings, and case sensitivity.

### CODE REVIEW

**Comment by nikkivaddepelli** [comment]
@bharat-dussa changes done

File: src/utils/string1/isStringEqual/isStringEqual.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export function isStringEqual(str1: string | null | undefined, str2: string | null | undefined): boolean {
+    if (str1 == null || str2 == null) {
+        throw new Error("Both parameters must be valid strings.");
+    }
+    
+    return str1 === str2;
```
[the remark this answers is not in the document]
Comment (nikkivaddepelli) [reply src/utils/string1/isStringEqual/isStringEqual.ts]: @coderabbitia please review

File: src/utils/string1/isStringEqual/isStringEqual.test.ts
Comment (bharat-dussa) [inline src/utils/string1/isStringEqual/isStringEqual.test.ts:1]: string1 ?

**Review by bharat-dussa — CHANGES_REQUESTED** [review]

**Comment by bharat-dussa** [comment]
Please raise a new PR since you have deleted HEAD Repo

**Pull request closed without merging**